src/main/scala/swiss/dasch/api/ExportEndpoint.scala

```diff
@@ -6,72 +6,61 @@
-
-    implicit val encoder: JsonEncoder[ExportResponse] = DeriveJsonEncoder.gen[ExportResponse]
-    implicit val schema: Schema[ExportResponse]       = DeriveSchema.gen[ExportResponse]
-  }
+import java.io.{ File, IOException }
 
-  private val shortcodePathVar = "shortcode"
+object ExportEndpoint {
+  private val shortcodePathVarName = "shortcode"
+  private type ContentDispositionStream = (ContentDisposition, ContentType, ZStream[Any, Nothing, Byte])
+  private val contentTypeApplicationZip = ContentType.parse("application/zip").toOption.get
+  private val downloadCodec             =
+    HeaderCodec.contentDisposition ++ HeaderCodec.contentType ++ ContentCodec.contentStream[Byte] ++ StatusCodec.status(
+      Status.Ok
+    )
 
-  private val exportEndpoint: Endpoint[String, ApiProblem, ExportResponse, None] = Endpoint
-    .post("export" / string(shortcodePathVar))
-    .out[ExportResponse](Status.Ok)
+  private val exportEndpoint: Endpoint[String, ApiProblem, ContentDispositionStream, None] = Endpoint
+    .post("export" / string(shortcodePathVarName))
+    .outCodec(downloadCodec)
     .outErrors(
       HttpCodec.error[ProjectNotFound](Status.NotFound),
       HttpCodec.error[IllegalArguments](Status.BadRequest),
       HttpCodec.error[InternalProblem](Status.InternalServerError),
     )
 
-  private val exportRoute: Routes[AssetService, ApiProblem, None] = exportEndpoint.implement { (shortcode: String) =>
-    ZIO
-      .fromEither(ProjectShortcode.make(shortcode))
-      .mapError(ApiProblem.invalidPathVariable(shortcodePathVar, shortcode, _))
-      .flatMap(AssetService.zipProject(_).some.mapError {
-        case Some(e) => ApiProblem.internalError(e)
-        case _       => ProjectNotFound(shortcode)
-      })
-      .map(ExportResponse.make)
-  }
-
-  val app = exportRoute.toApp
-
-  sealed trait ApiProblem
-  case class ProjectNotFound(shortcode: String)            extends ApiProblem
-  case class IllegalArguments(errors: Map[String, String]) extends ApiProblem
-  case class InternalProblem(errorMessage: String)         extends ApiProblem
-
-  object ApiProblem {
-    implicit val projectNotFoundEncoder: JsonEncoder[ProjectNotFound]   = DeriveJsonEncoder.gen[ProjectNotFound]
-    implicit val projectNotFoundSchema: Schema[ProjectNotFound]         = DeriveSchema.gen[ProjectNotFound]
-    implicit val illegalArgumentsEncoder: JsonEncoder[IllegalArguments] = DeriveJsonEncoder.gen[IllegalArguments]
-    implicit val illegalArgumentsSchema: Schema[IllegalArguments]       = DeriveSchema.gen[IllegalArguments]
-    implicit val internalErrorEncoder: JsonEncoder[InternalProblem]     = DeriveJsonEncoder.gen[InternalProblem]
-    implicit val internalErrorSchema: Schema[InternalProblem]           = DeriveSchema.gen[InternalProblem]
-
-    def internalError(t: Throwable): InternalProblem =
-      InternalProblem(t.getMessage)
+  private val handlerFun: String => ZIO[AssetService, ApiProblem, ContentDispositionStream] =
```

Comment:
`handlerFun` sounds very generic. Maybe something like `exportAssets` or `exportAssetsByShortcode` would be more descriptive of what it does? but since it's a private member with a very limited scope, I'm absolutely ok with `handlerFun` (or even `handler`, for that matter)

Reply:
renamed it to `postExportShortCodeHandler`